tests/suites/test_suite_psa_crypto_init.function

```diff
@@ -282,3 +282,73 @@ exit:
     mbedtls_psa_crypto_free( );
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PSA_CRYPTO_C */
```

Comment:
It depends how you look at it. It makes sense to group initialization-related tests together, whether they test initialization of the library or its data structures.

Given that we want to exercise the data structures, and those exercise functions are located in a different test suite, from a functional perspective (what works with what), I agree it makes sense to move the tests. I'd imagine hash context initialization tests would go into a hash test suite which also has hash functional tests, following this line of thought through to its logical conclusion. I'm not sure I see value in splitting hash calculation from hash operation tests as currently done, though.

Reply:
It makes sense to split calculation tests from validation and flow tests because calculation tests can be slow and are fairly boring and static most of the time whereas validation tests tend to change all the time. Whenever you've tweaked some validation or memory management somewhere, the validation/operation tests might reveal something interesting, whereas the calculation tests probably won't.

Reply:
Sure. I see this being an advantage when we run a subset of the tests (based on what was changed). Today, we always run the full suite, even when the files we change only affect a subset of our full suite.

Reply:
We always run the full suite on CI, and we should keep doing that, because it's very hard to tell for sure what positively definitely won't be impacted. Splitting into separate files is mainly useful for developer testing.